dashboard/src/actions/repos.ts

```diff
@@ -208,12 +208,25 @@ export const fetchRepos = (
       if (!otherNamespaces || !otherNamespaces.length) {
         dispatch(receiveRepos(repos.items));
       } else {
-        let totalRepos = repos.items;
+        const totalRepos = repos.items;
         await Promise.all(
           otherNamespaces.map(async otherNamespace => {
             dispatch(requestRepos(otherNamespace));
             const otherRepos = await AppRepository.list(currentCluster, otherNamespace);
-            totalRepos = totalRepos.concat(otherRepos.items);
+            // Avoid addiing duplicated repos: if two repos have the same uid, skip one.
```

Comment:
If two different namespaces have the same repo added, does that mean you won't see one of them? Why? From the PR description this appears to be related to the upgrade form (presenting a repo to choose), but in that case (which is upgrading in a specific namespace) I'd expect the only possibility is if the namespace has the same repo as a global repo? Are there other call-sites of this function that aren't related to upgrading? (like listing all repos) where we would want to see repos duplicated across different namespaces?

Reply:
This is comparing `uid` so the same repo in two different namespaces won't be identified as equals. Not sure why you need this code though, seems like an error requesting the same resources twice, which should not happen (so we should fix that).

Reply:
If I leave this part unchanged (in the upstream Kubeapps version) and then add a new repo as global, I get the following behavior:
![image](https://user-images.githubusercontent.com/11535726/100463790-b26ece80-30cc-11eb-8e64-f77eb6ea3818.png)
And the corresponding ` Encountered two children with the same key, 'kubeapps/jfrog'. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.`

I guess it is due to an unintended side effect when concatenating 
`AppRepository.list(currentCluster, namespace)` and `AppRepository.list(currentCluster, otherNamespace)` because they are global repos and they appear in both listings.

So, perhaps the best option so far is just to move this change to a separate PR and discuss it there.